packages/inference/src/HfInference.ts

```diff
@@ -653,10 +653,19 @@ export class HfInference {
 	): Promise<any> {
 		const mergedOptions = { ...this.defaultOptions, ...options };
 		const { model, ...otherArgs } = args;
+
+    const headers = {
+      Authorization: `Bearer ${this.apiKey}`,
+    }
+
+    if (mergedOptions.wait_for_model) {
```

Comment:
```suggestion
    if (options?.binary && mergedOptions.wait_for_model) {
```

From Narsil's comment